src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/VirtualProject.cs

```diff
@@ -30,22 +73,29 @@ internal static (string virtualProjectXml, bool isFileBasedProgram) MakeVirtualP
         var root = tree.GetRoot();
         var isFileBasedProgram = root.GetLeadingTrivia().Any(SyntaxKind.IgnoredDirectiveTrivia) || root.ChildNodes().Any(node => node.IsKind(SyntaxKind.GlobalStatement));
 
-        var virtualProjectXml = $"""
-            <Project Sdk="Microsoft.NET.Sdk">
+        var artifactsPath = GetArtifactsPath(documentFilePath);
 
+        var virtualProjectXml = $"""
+            <Project>
                 <PropertyGroup>
                     <OutputType>Exe</OutputType>
-                    <TargetFramework>net8.0</TargetFramework>
+                    <TargetFramework>net10.0</TargetFramework>
```

Comment:
My rationale for this change at this moment is:
- This project system is only going to work right with SDKs that can do `dotnet restore app.cs`. So, either net10 is going to be available or nothing is going to reliably work.
- Eventually we are delegating this problem to the SDK, there will be no notion of "what default TFM does IDE use." It's just whatever SDK tells us. So it is reasonable for us to do what is most expedient and meaningful for the immediate term.

Reply:
Going to take a quick pass to verify what will happen for prerelease extension users who do not have a net10 sdk installed.

Reply:
```suggestion
                    <TargetFramework>{isFileBasedProgram ? "net10.0" : "net8.0"}</TargetFramework>
```

Any reason to not do this? That way we're not risking your other concerns in the first place?

Reply:
I think the reason not to do this is: in any environment where net8.0 will work, with this system, net10.0 will also work. So why unload/load the project (changing TFM between net8 and net10) simply due to user including a top level statement or not.

Also, our next PR which we hope to get done in the next 2 weeks is going to completely delete this decision from our layer.

Reply:
I'm thinking the case where you have a non-filebased program, and simply don't have the 10.0 SDK installed. You (as we saw) might still get references even without a restore. You defintiely won't get them if you upgrade to the 10.0 SDK.

Reply:
Did you consider making this confugirable via environment variable? Basically if there is any question here and it's for the DEMO, could we give @DamianEdwards an out by assigning an environment variable with the overriden value.

Reply:
> I'm thinking the case where you have a non-filebased program, and simply don't have the 10.0 SDK installed. You (as we saw) might still get references even without a restore. You defintiely won't get them if you upgrade to the 10.0 TFM.

I'm sympathetic enough that we should try not to break and unbreak this arbitrarily.

This is something we can imagine will continue to work even in absence of an up to date SDK due to the strategy of using a "canonical project" to get references for the misc scenario.

I am just going to manual test it to make sure we are gracefully handling the change in TFM.

An env var also sounds like a good idea to fix it so a single value, in case changing it is messing up the project system state. I will introduce `DOTNET_RUN_FILE_TFM` for short term usage for that purpose.

Reply:
> I'm thinking the case where you have a non-filebased program, and simply don't have the 10.0 SDK installed. You (as we saw) might still get references even without a restore. You defintiely won't get them if you upgrade to the 10.0 TFM.

I think this case actually does not get better by changing TFM to net8.0 for "true loose files".

In a case where user *does* have net10 SDK, yet lacks net8 targeting pack on disk, specifying net8 in here will cause us to do the wrong thing. `dotnet restore app.cs` sees the TFM as net10, and will bring in net10 references, not net8 ones. So no references will light up for the user, but if we had targeted net10, then they would have lit up for that user.

Reply:
Basically in order to make "core library references for loose files" work properly, without relying on ambient state already being pre-restored onto the dev machine, we probably need to do the "canonical misc files project" idea we discussed earlier.

Right now what we have when the misc files project targets net8, is something which essentially only works if the user has warmed the caches up already. Which seems worse than requiring a user of the prerelease also install prerelease SDK in order for this feature to light up.

Reply:
> Right now what we have when the misc files project targets net8, is something which essentially only works if the user has warmed the caches up already. 

I guess my observation is that might work, or might not. But installing the pre-release SDK clearly won't solve it _either_ given we're having to fix restore in the first place.